SConstruct

```diff
@@ -54,10 +54,8 @@ TARGET_FLAGS = {
 def construct():
     ccflags = (
         ' -g -O2 -Wall -Wextra -Werror '
-        '-Wno-null-pointer-arithmetic '
```

Comment:
-Wno-null-pointer-arithmetic is needed on macOS:

src/tcp_connection.c:92:47: error: arithmetic on a null pointer treated as a cast from integer to pointer is a GNU extension [-Werror,-Wnull-pointer-arithmetic]
    return cp->cmsg_len - (size_t)(uintptr_t) CMSG_DATA((struct cmsghdr *) 0);
                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/socket.h:655:58: note: expanded from macro 'CMSG_DATA'
#define CMSG_DATA(cmsg)         ((unsigned char *)(cmsg) + \
                                 ~~~~~~~~~~~~~~~~~~~~~~~ ^